src/unif.ml

```diff
@@ -37,30 +37,44 @@ let instantiate : meta -> term array -> term -> bool = fun u env a ->
   let to_var t = match t with Vari v -> v | _ -> assert false in
   let b = Bindlib.bind_mvar (Array.map to_var env) (lift a) in
   Bindlib.is_closed b && (set_meta u (Bindlib.unbox b); true)
-    
+
+(** [unify_list l] tests equality between all the pairs of terms of [l], while
+    possibly instantiating metavariables. *)
+let rec unify_list : (term * term) list -> bool = fun l ->
+  match l with
+  | []                   -> true
+  | (a,b)::l when a == b -> unify_list l
```

Comment:
How often can a==b happen?

Reply:
Surprisingly often... ;)

I just tried to count this case on Matita, and this is used one million time after two minutes.

Reply:
For the whole Matita, the physical equality case is used `840 000 000` times!

Reply:
Are a and b just constants or bigger terms? Because we do no hashconsing...

Reply:
Hashconsing is not the only way for things to be equal. There is also substitution, and the instantiation of unification variables. In any case, this is an harmless optimisation.

Reply:
I see. That's interesting. So, I should add this in the pa branch too.